pytest_girder/pytest_girder/utils.py

```diff
@@ -1,16 +1,28 @@
 import asyncore
 import base64
 import cherrypy
+import contextlib
 import email
 import errno
+import inspect
 import json
 import os
 import six
 import smtpd
 import socket
+import subprocess
 import threading
 import time
 
+from girder.api import access
+from girder.api.describe import describeRoute, Description
+from girder.api.rest import Resource
+from girder.constants import ROOT_DIR, registerAccessFlag
+from girder.exceptions import RestException
+from girder.models.folder import Folder
+from girder.models.upload import Upload
+from girder.plugin import getPlugin
+from girder.utility.progress import ProgressContext
```

Comment:
In the past, we had to put girder imports inside functions in pytest_girder otherwise the coverage numbers were incorrect.  Is this still the case?

Reply:
You won't see it in a plugin.  This issue is that the import is executed *before* pytest starts tracking coverage.  Then when the test infrastructure to started up and these packages are imported by core, the cached module prevents the top-level code (and anything executed as a side-effect) from showing up as covered.

Reply:
Is this a documented limitation of pytest-cov? It seems like a pretty big hassle...

Reply:
Not really, we just have a weird use case.  We are using the library we are testing inside a pytest plugin that we are using to test it.

Reply:
We might be able to get around it by adding global fixture that deletes `sys.modules['girder']` before running the test.  I'm not sure what other consequences this might have.

Reply:
Oh I see, so this pattern would only be necessary for stuff in `pytest_girder`, not in the test files themselves. I'm fine with that, I will change these back.